src/http/client.rs

```diff
@@ -83,24 +84,35 @@ impl Client {
 
         let outgoing_body = OutgoingBody::new(AsyncOutputStream::new(wasi_stream), wasi_body);
 
-        struct IncomingResponseFuture {
-            subscription: WaitFor,
-            wasi: WasiFutureIncomingResponse,
+        pin_project! {
+            struct IncomingResponseFuture {
+                #[pin]
+                subscription: Option<WaitFor>,
+                wasi: WasiFutureIncomingResponse,
+            }
         }
         impl Future for IncomingResponseFuture {
             type Output = Result<Response<IncomingBody>>;
 
             fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
-                match pin_project(self.subscription).poll(cx) {
+                let this = self.project();
+                match this.subscription.as_pin_mut().expect("make it so").poll(cx) {
```

Comment:
Why is this option / expect here? Other than that, good to go!

Reply:
That was just me being confused. The subsequent patch fixes that now.